opensrp-core/src/test/java/org/smartregister/multitenant/check/EventClientSyncedCheckTest.java

```diff
@@ -74,15 +123,47 @@ public void isEventsClientSyncedShouldReturnFalseWhenEventClientRepositoryIsNull
+
+            networkUtilsMockedStatic.when(NetworkUtils::isNetworkAvailable).thenReturn(true);
+            coreLibraryMockedStatic.when(CoreLibrary::getInstance).thenReturn(coreLibrary);
+
+            Mockito.doReturn(opensrpContext).when(coreLibrary).context();
+            Mockito.doReturn(ApplicationProvider.getApplicationContext()).when(opensrpContext).applicationContext();
+            Mockito.doReturn(dristhiConfiguration).when(opensrpContext).configuration();
+            Mockito.doReturn(TEST_URL).when(dristhiConfiguration).dristhiBaseURL();
+            Mockito.doReturn(false).when(opensrpContext).hasForeignEvents();
+            Mockito.doReturn(eventClientRepository).when(opensrpContext).getEventClientRepository();
+            Mockito.doNothing().when(eventClientSyncedCheck).onSyncStart();
+            Mockito.doNothing().when(eventClientSyncedCheck).onSyncComplete(FetchStatus.fetchedFailed);
+            Mockito.doReturn(allSharedPreferences).when(opensrpContext).allSharedPreferences();
+            Mockito.doReturn("demo").when(allSharedPreferences).fetchRegisteredANM();
+            Mockito.doReturn("TeamA").when(allSharedPreferences).fetchDefaultTeam(ArgumentMatchers.anyString());
+            Mockito.doReturn(httpAgent).when(opensrpContext).getHttpAgent();
+            Mockito.doReturn(syncConfiguration).when(coreLibrary).getSyncConfiguration();
+            Mockito.doReturn(false).when(syncConfiguration).firebasePerformanceMonitoringEnabled();
+            Mockito.doReturn(false).when(syncConfiguration).disableSyncToServerIfUserIsDisabled();
+            Mockito.doReturn(true).when(httpAgent).verifyAuthorization();
+            Response<String> response = new Response<>(ResponseStatus.failure, "{}");
+            Mockito.doReturn(response).when(httpAgent).post(ArgumentMatchers.anyString(), ArgumentMatchers.anyString());
+            Mockito.doReturn(EasyMap.mapOf(AllConstants.KEY.EVENTS, eventJson)).when(eventClientRepository).getUnSyncedEvents(ArgumentMatchers.anyInt());
+
+            eventClientSyncedCheck.performPreResetAppOperations(DrishtiApplication.getInstance());
+
+            ArgumentCaptor<String> urlArgumentCaptor = ArgumentCaptor.forClass(String.class);
+            ArgumentCaptor<String> payloadArgumentCaptor = ArgumentCaptor.forClass(String.class);
+
+            // Verify that performSync() was called
+            Mockito.verify(httpAgent).post(urlArgumentCaptor.capture(), payloadArgumentCaptor.capture());
+
+            String requestBaseUrl = urlArgumentCaptor.getValue();
+            String requestPayload = payloadArgumentCaptor.getValue();
+
+            Assert.assertNotNull(requestBaseUrl);
+            Assert.assertNotNull(requestPayload);
+            Assert.assertEquals(expectedBaseUrlConstruct, requestBaseUrl);
+            Assert.assertEquals(expectedSyncPayloadConstruct, requestPayload);
```

Comment:
I think we can re-add the previous verifications here

```java
	Mockito.verify(eventClientSyncedCheck).onSyncStart();
        Mockito.verify(eventClientSyncedCheck).onSyncComplete(FetchStatus.fetchedFailed);
```

Reply:
This one may need to be followed up separately if we want to keep the original test case implementation. The issue is we need a way to trigger the broadcast during test runtime and capture the _Sync Status Events_ for verification.

I had updated so that for now we are checking whether the Sync was called by verifying an invocation to the `httpAgent` POST method after the `performPreResetAppOperations` call. 

If this doesn't suffice for this test case we can create a new ticket to address reverting to the old approach.